sdk/core/azure-core-serializer-json-jackson/src/test/java/com/azure/core/serializer/json/jackson/implementation/GeoJsonDeserializerTests.java

```diff
@@ -42,21 +42,21 @@ public class GeoJsonDeserializerTests {
     @Test
     public void jsonWithoutTypeThrows() {
         String missingType = "{\"coordinates\":[0,0]}";
-        Assertions.assertThrows(IllegalStateException.class,
+        Assertions.assertThrows(IOException.class,
```

Comment:
@JonathanGiles, @srnagar, @lmolkova this was changed as I realized/found the deserialization, and serialization, handling for `JsonSerializable` in `azure-core` try/catch'd in a way where RuntimeExceptions were wrapped in IOExceptions:

https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/implementation/ReflectionSerializable.java#L196

Thoughts on changing this to allow `RuntimeException`s to be thrown without wrapping.